src/components/Fetch/index.js

```diff
@@ -169,15 +169,13 @@ export class Fetch extends React.Component {
           bookmarkStr += ',';
         }
       }
-      searchParams.append('bookmarks', bookmarkStr);
+      parsed.bookmarks = bookmarkStr;
     }
     if (this.state.server) {
-      searchParams.append('server', this.state.server);
+      parsed.server = this.state.server;
     }
-    this.props.history.push({
-      pathname: nextUrl,
-      search: searchParams.toString()
-    });
+    /* global global:{} */
+    global.window.location.hash = queryString.stringify(parsed);
```

Comment:
I know I said otherwise earlier, but could you try removing the comment, and changing this to "window.location"? That should eliminate the lint error. If it does, go with that, otherwise leave this as is.

Reply:
Followup: Let's use the history api to fix the back button even when we do this.
(`history.replaceState`)